### PR TITLE
add: dockerhub extension

### PR DESCRIPTION
**Pull request by k8scat**
## Description

Search images and tags in DockerHub

## Type of change

New extension

## Screencast

image

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/dockerhub/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "description": "Search images on DockerHub.",
+  "icon": "docker-icon.png",
+  "author": "k8scat",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "search-image",
+      "title": "Search Images",
+      "subtitle": "Images on Docker Hub",
+      "description": "Search images on Docker Hub",
```
Comment (fe9lix) [inline extensions/dockerhub/package.json:14]: I'd suggest to use "Search Docker Images" as title and leave the subtitle empty. Also see our store guidelines: https://developers.raycast.com/basics/prepare-an-extension-for-store

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @k8scat, thanks for the contribution! I've left a comment re command title and subtitle. 
Another suggestion: is there a default list you could load (latest, most popular, ...)? Currently the default state is an empty list, which is not the nicest experience for end users.

**Comment by k8scat** [comment]
Is there a best practice for the list pagination instead showing all items?

**Comment by fe9lix** [comment]
> Is there a best practice for the list pagination instead showing all items? @fe9lix

Nope, list pagination is not yet supported. You can group items into sections in a list. For now, it's best to only load a subset (list should not contain more than 1000 entries) and then use search to narrow it down. If you want to support different lists, the best way is to create a separate command for it.

**Comment by fe9lix** [comment]
Thanks for the updates. Just tested again and I run into an error when I open the command and then type "u" into the search bar. I get a Toast with "Search failed" and "Cannot read properties of null (reading 'map')"

**Comment by k8scat** [comment]
> Thanks for the updates. Just tested again and I run into an error when I open the command and then type "u" into the search bar. I get a Toast with "Search failed" and "Cannot read properties of null (reading 'map')"

Fixed. And this pr is done if there is no problem.

**Comment by fe9lix** [comment]
Thanks for the update! Two more things I've noticed:
- I get duplicate keys warnings in the console from React – probably coming from the key property of some list items (key should have a stable identity and be assigned to the top-level component in case the list item is wrapped in another component)
- Please use the action form for action panel items, i.e. "Tags" should be "Show Tags"...

**Comment by k8scat** [comment]
> * use the action form for action panel items

What does `use the action form for action panel items` mean? Is that suggest to use `SubmitFormAction` instead of `PushAction`?

**Comment by fe9lix** [comment]
> > * use the action form for action panel items
> 
> What does `use the action form for action panel items` mean? Is that suggest to use `SubmitFormAction` instead of `PushAction`?

Sorry, no, I meant grammatically: "Show Tags" (action) instead of just "Tags" (Noun), also see the store guidelines on this: https://developers.raycast.com/basics/prepare-an-extension-for-store

**Comment by k8scat** [comment]
> > > * use the action form for action panel items
> > 
> > 
> > What does `use the action form for action panel items` mean? Is that suggest to use `SubmitFormAction` instead of `PushAction`?
> 
> Sorry, no, I meant grammatically: "Show Tags" (action) instead of just "Tags" (Noun), also see the store guidelines on this: https://developers.raycast.com/basics/prepare-an-extension-for-store

Already fixed.

**Review by fe9lix — APPROVED** [review]

**Pull request merged**